honeybee_energy/schedule/ruleset.py

```diff
@@ -1169,6 +1197,20 @@ def _get_avg_week(name, schedules, weights, timestep_resolution, rule_indices):
             avg_mtx[5], avg_mtx[6], avg_mtx[7], timestep_resolution,
             schedules[0].schedule_type_limit, avg_mtx[8], avg_mtx[9])
 
+    @staticmethod
+    def _instance_in_array(object_instance, object_array):
+        """Check if a specific object instance is already in an array.
+
+        This can be much faster than  `if object_instance in object_arrary`
```

Comment:
I believe you have done your research on this. At the same time does this logic mean that the way `==` is implemented in the objects that check hasn't been implemented efficiently? I don't mean to go back and revise them all right now but just something to think about.

Reply:
It's not that the method is implemented inefficiently.  It's that schedules have a lot of properties to check in order to know that they are equal to one another.  If the object in question was something as simple as a material, all that we need to check in order to know if the materials are equal is the name and a handful of properties. So this is a fast check.  But for ScheduleRulesets, there are often several ScheduleDays to check and we need to check that all of these ScheduleDay values and times are equal to one another. There's also ScheduleTypelimits and other things to check like rules. So it takes significantly longer than just checking if something is the same instance.  Also, with these changes I made in this PR, the intention is that  someone uses the same ScheduleDay instance when they have a cross-referenced ScheduleDay (not two different ScheduleDay instances that are equivalent to one another). So this method here actually enforces this intention and helps discourage the latter.